rascal-vscode-extension/src/util/PositionConverter.ts

```diff
@@ -0,0 +1,189 @@
+
+/**
+ * This class allows to convert text locations between VS Code and Rascal.
+ * Some characters, which VS Code counts as two characters because it uses UTF-16 encoding,
+ * are only a single character for Rascal, which uses UTF-32 encoding.
+ *
+ * Assuming UTF-16 encoding, because this code is executed in TypeScript,
+ * all methods in this class therefore iterate character-wise over a text to find pairs of characters
+ * that would be encoded as a single character in UTF-32.
+ *
+ * If they find such a pair, they change the position to encount for the difference in encoding:
+ * If the position comes from Rascal, it is too short and must be increased by 1.
+ * If the position comes from VS Code, it is too long and must be decreased by 1.
+ */
+export class PositionConverter {
+
+    /***************************************
+     *              Rascal -> VS Code      *
+     ***************************************/
+
+    /**
+     * Converts the column position from Rascal (UTF-32) to VS Code (UTF-16).
+     * @param td the text document where the column information is located in.
+     * @param line the line in which the column to be changed is located.
+     * @param rascalColumn the column as given by Rascal.
+     * @returns the column as understood by VS Code.
+     */
+    static rascalToVSCodeColumn(td: vscode.TextDocument, line: number, rascalColumn: number): number {
+        const fullLine = td.lineAt(line).text;
+        let result = rascalColumn;
+        for (let i = 0; i < fullLine.length && i < result; i++) {
+            const c = fullLine.charCodeAt(i);
+            if (PositionConverter.isHighSurrogate(c) && (i + 1) < fullLine.length && PositionConverter.isLowSurrogate(fullLine.charCodeAt(i + 1))) {
+                i++;
+                result++;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Converts the offset and length position from Rascal (UTF-32) to VS Code (UTF-16).
+     * @param td the text document where the information is located in.
+     * @param offset the offset as given by Rascal.
+     * @param length the length as given by Rascal.
+     * @returns the offset and length as understood by VS Code.
+     */
+    static rascalToVSCodeOffsetLength(td: vscode.TextDocument, offset: number, length: number): [number, number] {
+        const fullText = td.getText();
+        let endOffset = offset + length;
+        for (let i = 0; i < fullText.length && i < endOffset; i++) {
+            const c = fullText.charCodeAt(i);
+            if (PositionConverter.isHighSurrogate(c) && (i + 1) < fullText.length && PositionConverter.isLowSurrogate(fullText.charCodeAt(i + 1))) {
+                if (i <= offset) { // the character comes before the offset, so it must shift the offset
+                    offset++;
+                }
+                endOffset++;
+                i++;
+            }
+        }
+        return [offset, endOffset];
```

Comment:
this has a bug, as the endOffset is not the length. Maybe should it be `endOffset - offset` ? or we should make it clear it returns 2 positions.